packages/i18n-utils/src/locale-context.tsx

```diff
@@ -26,7 +29,29 @@ export const LocaleProvider: React.FC< Props > = ( { children, localeSlug } ) =>
  * }
  */
 export function useLocale(): string {
-	return useContext( localeContext );
+	const fromProvider = useContext( localeContext );
+
+	function getWpI18nLocaleSlug() {
+		return i18n.getLocaleData()?.[ '' ]?.language;
+	}
+
+	const [ fromWpI18n, setWpLocale ] = useState( getWpI18nLocaleSlug() );
+
+	useEffect( () => {
+		// If the <LocaleProvider> has been used further up the component tree
+		// then we don't want to subscribe to any defaultI18n changes.
+		if ( fromProvider ) {
+			return undefined;
+		}
+
+		setWpLocale( getWpI18nLocaleSlug() );
+
+		return i18n.subscribe( () => {
+			setWpLocale( getWpI18nLocaleSlug );
```

Comment:
This was just a typo.

> passing it as a callback

😮 good catch! It took me a while to figure out how the tests were passing 😅

Reply:
Yes, they both work because `setState` accepts either a value or a value-returning callback 🙂 

I liked this article about the risks of callbacks: https://jakearchibald.com/2021/function-callback-risks/